### ISSUE TITLE
New layout suggestion: Masonry

### ISSUE THREAD

**Issue by IjzerenHein** [IjzerenHein/famous-flex#57]

**Comment by IjzerenHein** [comment]
> @matt-d-rate wrote:

Masonry layout (similar to this: http://masonry.desandro.com), which is like a grid, but supports widgets which can be of variable height and perhaps variable width.

**Comment by IjzerenHein** [comment]
> @grimen wrote:

With respect to custom widths too is a quite common scenario: http://erikjohanssonphoto.com/work/imagecats/personal

**Comment by IjzerenHein** [comment]
> @matt-d-rat wrote:

Agreed, it would need to support variable height and widths for each of the items in the grid.

**Comment by matt-d-rat** [comment]
Super excited, can't wait to try this out!

**Comment by IjzerenHein** [comment]
It might be a while before I have the time to work on this. It would basically be a modified version of CollectionLayout. If anyone wants to help, that would be great. Let me know and I will you give the support you need.

**Comment by grimen** [comment]
For the record: https://github.com/nmn/famousry

**Comment by Qvatra** [comment]
I was trying to write Masonry like layout with only difference - height of outer container is fixed.
here is what I have: http://codepen.io/Qvatra/pen/zxRyVy
hope this will help

**Comment by Qvatra** [comment]
check out the following Gist with a layout that I wrote for your LayoutController: https://gist.github.com/Qvatra/65d481ce0596e6f1b012

here is codePen example: http://codepen.io/Qvatra/pen/ogmYJW

**Comment by mhwheeler** [comment]
+1. This is a very useful contemporary style.

**Comment by bardu** [comment]
Will masonry style layout ever be included to your library?

BTW, it is a joy working with famous-flex, even if we don't know what's happening at Famous Industries Inc.

**Comment by IjzerenHein** [comment]
It's not planned for the foreseeable future at this moment. It would be really awesome if the community could help to build this layout :)
My focus is currently on building the next version of famous-flex for mixed-mode (or for possible forks , and if achievable even for famous v0.3). It's a lot of work, but I can assure you, it will be immensely awesome :) I hope I can share some demo's with everyone soon.

**Comment by bardu** [comment]
Can't wait for it.
Regarding forks, how about [SamsaraJS](https://github.com/dmvaldman/samsara)?

**Comment by IjzerenHein** [comment]
Yes that one potentially too.
